packages/context/src/resolver.ts

```diff
@@ -273,9 +277,9 @@ export function resolveInjectedProperties(
   return resolveMap(injectedProperties, (injection, p) => {
     if (!injection.bindingSelector && !injection.resolve) {
       const name = getTargetName(constructor, p);
-      throw new Error(
-        `Cannot resolve injected property ${name}: ` +
-          `The property ${p} is not decorated for dependency injection.`,
+      throw new ResolutionError(
+        `The binding selector or resolve function is missing for injection of property '${name}'`,
```

Comment:
I have no idea what does it mean that a binding selector is missing or a resolve function is missing. The terms "bindingSelector" and "resolve" are speaking from the implementation details, which most LB4 users don't understand. (I don't know what do they mean either without reading the surrounding code here.)

Let's find a way how to describe this error using the terms that LB users will understand.

I think we should differentiate between the situation when the user forgot to apply `@inject()` and the situation where a decorator was applied but did not provide enough information to resolve the injected value.

Example error message for the first situation:

> The property '${name}' is not decorated for dependency injection. Did you forget to call @inject() decorator to configure dependency injection?

(Can this actually happen? Do we require any properties to be decorated for DI?)

In the second case, I think it's usually a problem in the inject-like decorator implementation that the user cannot do much about, is that right? If yes, then the message should make that clear.

For example:

> The property '${name}' is decorated for dependency injection, but the decorator did not provide enough information to resolve the value to be injected. Either `bindingSelector` or `resolve` is required.

Reply:
The only case for this error is:

`@inject('', { /* no resolve in options */}`

I think it's better to detect such problem at `@inject` decorator level.

Reply:
I improved the code to throw when `@inject` is called.